src/helix-ui/utils/alignment.js

```diff
@@ -233,43 +233,63 @@ export function normalizePosition (position) {
 export function optimizePositionForCollisions (position, collides) {
     let { xAlign, yAlign } = getAlignment(position);
 
-    // COLLIDE TOP
-    // 'top-*' -> 'bottom-*' (CHANGE)
-    // '{H}-top' -> '{H}-bottom' (CHANGE)
-    // 'bottom-*' -> 'bottom-*' (no change)
-    // '{H}-bottom' -> '{H}-bottom' (no change)
-    // '{H}-start|middle|end' -> '{H}-start|middle|end' (no change)
-    if (collides.top && yAlign === 'top') {
+    // ----- COLLIDE WITH TOP EDGE -----
+    // CHANGE
+    // - 'top-*'            -> 'bottom-*'
+    // - '(left|right)-top' -> '(left|right)-bottom'
+    // - '(left|right)-end' -> '(left|right)-start'
+    //
+    // IGNORE
+    // - 'bottom-*'
+    // - '{H}-bottom'
+    // - '{H}-start'
+    // - '{H}-middle'
+    if (collides.top && yAlign.match(/top|end/)) {
```

Comment:
When a top collision occurs, we have to invert the vertical component of the position when it's Y alignment is either `top` (menu height grows upward from x-axis of target, with additional content) or `end` (menu height grows upward from bottom of target, with additional content).